swarm/src/lib.rs

```diff
@@ -1118,13 +1118,16 @@ where
                 self.pending_handler_event = Some((peer_id, handler, event));
             }
             ToSwarm::NewExternalAddrCandidate(addr) => {
+                // Unless we have listeners, all candidates are ephemeral and are not reachable externally
+                if self.listeners().next().is_none() {
+                    return;
```

Comment:
I wonder if this is worth a log statement?

```suggestion
					tracing::debug!(%addr, "Discarding external address candidate because we have no listeners");
                    return;
```

Reply:
I think so, is debug a good level for this?

Reply:
Would you rather have it on `trace`? I think `info` or above could be very noisy for clients that never open listeners.

Reply:
I was not sure what level you wanted because I didn't see the suggestion itself somehow. Applied.